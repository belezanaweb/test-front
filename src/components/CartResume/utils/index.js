export const normalizeCartData = data => {
  return {
    productsQuantity: data.items.length,
    subTotal: data.subTotal.toFixed(2).replace('.', ','),
    shipping: data.shippingTotal.toFixed(2).replace('.', ','),
    discount: data.discount.toFixed(2).replace('.', ','),
    total: data.total.toFixed(2).replace('.', ','),
  };
};
