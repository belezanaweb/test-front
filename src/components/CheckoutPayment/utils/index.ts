import * as Yup from 'yup'

const errors = {
  name: 'Insira um nome válido',
  code: 'Insira um código válido',
  number: 'Insira um número válido',
  expiration: 'Insira uma data válida',
}

export const initialValues = {
  creditCardName: '',
  creditCardNumber: '',
  creditCardExpiration: '',
  creditCardCode: '',
}

const regexOnlyNumbers = /[^\d.-]+/g

export const CheckoutPaymentSchema = Yup.object().shape({
  creditCardName: Yup.string()
    .min(2, errors.name)
    .matches(/^([A-Za-z\u00C0-\u00D6\u00D8-\u00f6\u00f8-\u00ff\s]*)$/gi, errors.name)
    .required(errors.name),
  creditCardNumber: Yup.string()
    .min(2, errors.number)
    .test('len', errors.number, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 16
    })
    .required(errors.number),
  creditCardExpiration: Yup.string()
    .min(2, errors.name)
    .test('len', errors.expiration, (val) => {
      const value = val.replace(regexOnlyNumbers, '')

      return value.length === 4
    })
    .required(errors.expiration),
  creditCardCode: Yup.string()
    .matches(/^[0-9]+$/, errors.code)
    .test('len', errors.code, (val) => val.length === 3)
    .required(errors.code),
})
