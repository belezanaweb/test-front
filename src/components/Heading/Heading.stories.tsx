import React from 'react';

import theme, { Theming } from '../../theme';

import Box from '../Box';

import Heading from './Heading';

const Provider = (storyFn: () => React.ReactNode) => (
  <Theming theme={theme}>
    {storyFn()}
  </Theming>
);

export default {
  component: Heading,
  decorators: [Provider],
  title: 'Heading',
};

export const Default = () => (
  <Box css={{ padding: '20px' }}>
    <Heading>The quick brown fox jumps over the lazy dog</Heading>
  </Box>
);
