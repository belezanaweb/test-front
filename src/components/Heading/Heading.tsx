import styled from 'styled-components';

import Text from '../Text';

const Heading = styled(Text)`
  color: ${({ theme }) => theme.colors.grayDark};
  display: inline-flex;
  font-size: 14px;
  font-weight: 700;
  line-height: 16px;
  margin: 0;
  padding: 10px 10px 6px;
  text-transform: uppercase;
`;


export default Heading;