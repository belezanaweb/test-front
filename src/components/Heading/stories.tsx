import React from 'react'
import { Story } from '@storybook/react/types-6-0'
import { Heading, HeadingProps } from '.'

export default {
  title: 'Heading',
  component: Heading
}

export const Basic: Story<HeadingProps> = args => <Heading {...args} />

Basic.args = {
  title: 'Produtos'
}
