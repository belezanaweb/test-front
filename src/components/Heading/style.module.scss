@import "styles";

.heading {
    color: $c-text-heading;
    font-size: rem(14);
    font-weight: bolder;
    padding: 0.2em;
    margin-top: 0.5em;
    text-align: left;
    text-transform: uppercase;
    width: 100%;
    white-space: nowrap;
    @include device(desktop) {
        font-size: rem(16);
    }
}
