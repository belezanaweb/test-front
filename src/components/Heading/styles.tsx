import styled, { css } from 'styled-components'

export const Wrapper = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.size.font14};
    font-weight: ${theme.font.weight.bold};
    line-height: ${theme.font.lineHeight.lh17};
    color: ${theme.color.title};
    text-transform: uppercase;
  `}
`
