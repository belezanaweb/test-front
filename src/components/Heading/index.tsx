import React from 'react'
import * as S from './styles'

export type HeadingProps = {
  title: string
}

export const Heading: React.FC<HeadingProps> = ({ title }) => {
  return <S.Wrapper>{title}</S.Wrapper>
}
