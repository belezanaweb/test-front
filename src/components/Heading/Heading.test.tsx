import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { provider } from '../../utils/testUtils';

import Heading from './Heading';

describe('<Heading />', () => {
  it('should render Heading', () => {
    const { container } = provider(<Heading>The quick brown fox jumps over the lazy dog</Heading>, render);

    expect(container).toMatchSnapshot();
  });
});
