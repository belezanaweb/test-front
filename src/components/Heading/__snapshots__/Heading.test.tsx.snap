// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render Heading 1`] = `
<div>
  <span
    class="sc-AxjAm sc-AxirZ emCqti"
  >
    The quick brown fox jumps over the lazy dog
  </span>
</div>
`;
